src/node_nanomsg.cc

```diff
@@ -125,6 +130,41 @@ NAN_METHOD(Getsockopt) {
     NanReturnValue(obj);
 }
 
+NAN_METHOD(Setopt) {
+  NanScope();
+
+  int level = args[1].integer;
+  int option = args[2].integer;
+
+  if(option == NN_SOCKET_NAME){
+    utf8 str(args[3]);
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, *str, str.length())));
+  } else {
+    int optval = args[3].integer;
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, &optval, sizeof(optval))));
```

Comment:
One issue I just realized with this approach is that we always treat optval as a signed integer, but in fact `NN_LINGER` [can be negative](http://nanomsg.org/v0.2/nn_setsockopt.3.html)!
```
Negative value means infinite linger.
```